tests/test_client_functional.py

```diff
@@ -632,6 +632,28 @@ async def handler(request: web.Request) -> web.Response:
     assert txt == "Test message"
 
 
+async def test_ssl_client_alpn(
+    aiohttp_server: AiohttpServer,
+    aiohttp_client: AiohttpClient,
+    ssl_ctx: ssl.SSLContext,
+) -> None:
+
+    async def handler(request: web.Request) -> web.Response:
+        sslobj = request.transport.get_extra_info("ssl_object")
+        assert sslobj.selected_alpn_protocol() == "http/1.1"
+        return web.Response(text="Test message")
+
+    app = web.Application()
+    app.router.add_route("GET", "/", handler)
+    ssl_ctx.set_alpn_protocols(["http/1.1"])
+    server = await aiohttp_server(app, ssl=ssl_ctx)
```

Comment:
Do we need this on the server side? I thought this was defined entirely by the client..

Reply:
alpn is also defined on server side. The protocol selected by alpn is based on alpn ext in both server's and client's hello. We need to set alpn on server side first and see if we can get selected protocol to test alpn ext on client side.

Reply:
Right, I think I misunderstood. If we don't define it, then I guess it just ignores the extension and continues with a normal handshake.